sui_types/src/batch.rs

```diff
@@ -3,21 +3,34 @@
 
 use crate::base_types::{AuthorityName, TransactionDigest};
 use crate::crypto::{sha3_hash, AuthoritySignature, BcsSignable};
+use crate::messages::TransactionInfoResponse;
 use serde::{Deserialize, Serialize};
 
 pub type TxSequenceNumber = u64;
 
 /// Either a freshly sequenced transaction hash or a batch
-#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub enum UpdateItem {
-    Transaction((TxSequenceNumber, TransactionDigest)),
+    Transaction(
+        (
+            TxSequenceNumber,
+            TransactionDigest,
+            Option<Box<TransactionInfoResponse>>,
+        ),
+    ),
     Batch(SignedBatch),
 }
 
 pub type BatchDigest = [u8; 32];
 
 #[derive(Eq, PartialEq, Ord, PartialOrd, Clone, Hash, Default, Debug, Serialize, Deserialize)]
-pub struct TransactionBatch(Vec<(TxSequenceNumber, TransactionDigest)>);
+pub struct TransactionBatch(
+    pub  Vec<(
+        TxSequenceNumber,
+        TransactionDigest,
+        Option<Box<TransactionInfoResponse>>,
```

Comment:
I wonder if it would be better to have the TransactionInfoResponse just be a separate Vec.